workflow/executor/k8sapi/client.go

```diff
@@ -129,6 +129,14 @@ func (c *k8sAPIClient) until(ctx context.Context, f func(pod *corev1.Pod) bool)
 					}
 					pod, ok := event.Object.(*corev1.Pod)
 					if !ok {
+						// If it's not a pod, try convert to status first
+						// and exit the loop if its status is failure.
+						status, ok := event.Object.(*metav1.Status)
+						if ok {
+							if status.Status == "Failure" {
+								return true, fmt.Errorf(status.Message)
```

Comment:
Not necessarily for the bug I reported earlier but this avoids another case that would make the wait container stuck here. In case of status that is failure, we should return `true` here to exit early instead of keep retrying.

Reply:
I think you can just change the next line to return true?

Reply:
Yes, in this case actually we don't care whether it's success or failure status since we are only performing pod watch here. I'll remove/update this part.